packages/warehouses/src/warehouseClients/BigqueryWarehouseClient.ts

```diff
@@ -294,13 +296,30 @@ export class BigqueryWarehouseClient extends WarehouseBaseClient<CreateBigqueryC
         const datasetTablesResponses = await Promise.all(
             datasets.map((d) => d.getTables()),
         );
-        return datasetTablesResponses.flatMap(([tables]) =>
-            tables.map((t) => ({
-                database: t.bigQuery.projectId,
-                schema: t.dataset.id!,
-                table: t.id!,
-            })),
+        const tablesWithPartitions = await Promise.all(
+            datasetTablesResponses.flatMap(([tables]) =>
+                tables.map(async (t) => {
+                    const [metadata] = await t.getMetadata();
+                    const partitionColumn: PartitionColumn =
+                        metadata.timePartitioning || metadata.rangePartitioning
```

Comment:
this is where we extract the metadata info from bigquery

Reply:
This is parallel API calls one per table right? IF the user has 300 tables, it's 300 API calls. Did you stress test this yet?

Reply:
Good point, I didn't stress test this, We have 14 tables on bigquery, and I didn't notice this was taking longer. 

Instead of making 1 query per table, I can make 1 query per schema, running this custom code. I don't see this information on the schema metadata

![image](https://github.com/user-attachments/assets/d77ea518-1650-40e6-af2f-09ddf06f20bd)

Reply:
This query doesn't return the `range` or date granularity, which is useful, but perhaps less important than performance, what do you think ?